cmd/containerboot/main.go

```diff
@@ -305,6 +324,35 @@ authLoop:
 	}
 }
 
+func configureForwarding(ctx context.Context, client *tailscale.LocalClient, cfg *settings) error {
+	if cfg.ProxyTo == "" {
+		return nil
+	}
+	st, err := client.StatusWithoutPeers(ctx)
+	if err != nil {
+		return err
+	}
+	if len(cfg.FunnelTCPPorts) == 0 {
+		return installIPTablesRule(ctx, cfg.ProxyTo, st.Self.TailscaleIPs)
+	}
+	if len(st.CertDomains) == 0 {
+		return errors.New("no cert domains, cannot configure TCP forwarding")
+	}
+	cd := st.CertDomains[0]
+	sc := &ipn.ServeConfig{
+		AllowFunnel: make(map[ipn.HostPort]bool),
+		TCP:         make(map[uint16]*ipn.TCPPortHandler),
+	}
+	for f, t := range cfg.FunnelTCPPorts {
+		sc.TCP[f] = &ipn.TCPPortHandler{
+			TCPForward:   fmt.Sprintf("%s:%d", cfg.ProxyTo, t),
+			TerminateTLS: cd,
+		}
+		sc.AllowFunnel[ipn.HostPort(fmt.Sprintf("%s:%d", cd, f))] = true
+	}
+	return client.SetServeConfig(ctx, sc)
```

Comment:
How do we know when funnel is working? Ideally the operator would like some signal that pod startup has completed correctly, so it doesn't advertise the ingress hostname until it's actually serving.